### PR TITLE
all benchmarks working

### PR DESCRIPTION
**Pull request by jano1906**

### CODE REVIEW

File: jointformer/models/utils.py [hunk 1]
```diff
@@ -26,15 +26,18 @@ def __init__(self, *args, **kwargs):
...
+            if _is_cls_token_embedding:
+                attn_mask = attn_mask[:, 1:]
+            w = attn_mask / attn_mask.sum(dim=-1, keepdim=True)
             w = w.unsqueeze(-2)
-            global_embedding = w @ self['cls_embeddings']
+            global_embedding = w @ self['lm_embeddings']
             return global_embedding.squeeze(-2)
         else:
-            return self['cls_embeddings'].mean(dim=-1)
+            assert False
```
Comment (jano1906) [inline jointformer/models/utils.py:40]: [WARNING] this was done to extract lm features from jointformer

File: experiments/joint_training/features_eval.sh [hunk 2]
```diff
@@ -31,7 +31,8 @@ for model in ${models[@]}; do
             --output_dir=$output_dir \
             --target=$target \
             --lr=$lr \
-            --evals ${evals[@]}
+            --evals ${evals[@]} \
+            --epochs 0
```
Comment (jano1906) [inline experiments/joint_training/features_eval.sh:35]: [Warning] this should be reverted

File: experiments/joint_training/features_eval.py [hunk 3]
```diff
@@ -171,13 +195,17 @@ def main(args):
...
 
     for eval_str in args.evals:
-        out_dir = os.path.join(args.output_dir, eval_str, args.target, "_".join(str(datetime.now()).split()))
-        os.makedirs(out_dir, exist_ok=False)
+        path_prefix = os.path.join(args.output_dir, eval_str, args.target)
+        out_dir = os.path.join(path_prefix, os.listdir(path_prefix)[0])
+        #out_dir = os.path.join(path_prefix, "_".join(str(datetime.now()).split()))
+        #os.makedirs(out_dir, exist_ok=False)
+        writer = SummaryWriter(out_dir)
         if eval_str == "eval_linear":
```
Comment (jano1906) [inline experiments/joint_training/features_eval.py:203]: [WARNING] this should be reverted

File: experiments/joint_training/assess_distribution_learning.sh [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
-#        --output outputs/output.js
-
-python3 experiments/joint_training/assess_distribution_learning.py \
-        --path_to_model_ckpt /home/jano1906/git/jointformer/checkpoints/GNN_Edge_MLP_MoLeR__2022-02-24_07-16-23_best.pkl \
-        --path_to_model_config configs/models/moler \
-        --path_to_task_config configs/tasks/guacamol/physchem/ \
-        --chembl_training_file ~/datasets/guacamol/guacamol_v1_train.smiles \
-        --output outputs/output.js
\ No newline at end of file
+#        --output outputs/output.js
```
Comment (jano1906) [inline experiments/joint_training/assess_distribution_learning.sh:16]: This can be deleted

File: experiments/data/extract_targets.sh [hunk 5]
```diff
@@ -1,20 +1,17 @@
 #!/bin/bash
 
-splits=("train" "val" "test")
-targets=("plogp" "qed" "guacamol_mpo" "physchem")
+splits=(train) #splits=("train" "val" "test")
+targets=(guacamol_mpo) #targets=("plogp" "qed" "guacamol_mpo" "physchem")
```
Comment (jano1906) [inline experiments/data/extract_targets.sh:4]: This should be reverted

File: jointformer/models/jointformer.py [hunk 6]
```diff
@@ -39,7 +39,8 @@ def __init__(
             num_prediction_tasks: int,
             num_physchem_tasks: Optional[int] = DEFAULT_NUM_PHYCHEM_TASKS,
             init_weights: bool = True,
-            tie_weights: bool = True
+            tie_weights: bool = True,
+            set_separate_task_tokens = False,
```
Comment (AdamIzdebski) [inline jointformer/models/jointformer.py:43]: There should be no `set_separate_task_token` arg

**Review by AdamIzdebski — APPROVED** [review]

**Pull request merged**